src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1647,6 +1650,19 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxLis
                     _pool.Free(constraints);
                 }
             }
+
+            SyntaxToken? eatRecordModifierIfAvailable()
+            {
+                Debug.Assert(keyword.Kind == SyntaxKind.RecordKeyword);
+                if (CurrentToken is { Kind: SyntaxKind.ClassKeyword or SyntaxKind.StructKeyword })
+                {
+                    var result = EatToken();
+                    result = CheckFeatureAvailability(result, MessageID.IDS_FeatureRecordStructs);
+                    return result;
+                }
+
+                return null;
```

Comment:
fwiw, i would just inline this method.

Reply:
The method is pretty crowded, so I'm extracting what's possible.